### PR TITLE
combine low and high token id

### PR DESCRIPTION
**Pull request by manlikeHB**
- [ ] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)

Combination of the NFTs high and low token id hex value into a single Integer value.

### CODE REVIEW

File: backend/routes/indexer/combineTokenId.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package indexer
+
+import (
+	"math/big"
+)
+
+
+func combineLowAndHighTokenId(tokenIdLowInt, tokenIdHighInt int64) (*big.Int) {
+	// Convert int64 parts to big.Int
+	low :=  new(big.Int).SetInt64(tokenIdLowInt)
```
Comment (b-j-roberts) [inline backend/routes/indexer/combineTokenId.go:10]: If we parse low and high as int64, there will be 64 bits of data chopped off of each of the low and high value on the LHS.

Each low and high are encoded as left padded u128. I think the best way to create this function would be to take the low and high as their hex string, then properly combine the strings while removing the left padding 0s. Then you can interpret the string as u256 ( big.Int should have a from string like function )

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: backend/routes/indexer/combineTokenId.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package indexer
+
+import (
+	"math/big"
+)
+
+
+func combineLowAndHighTokenId(tokenIdLowInt, tokenIdHighInt int64) (*big.Int) {
```
[the remark this answers is not in the document]
Comment (manlikeHB) [reply backend/routes/indexer/combineTokenId.go:8]: Alright.

File: backend/routes/indexer/combineTokenId.go (hunk 1 above)
Comment (manlikeHB) [reply backend/routes/indexer/combineTokenId.go:10]: Noted.

File: backend/routes/indexer/utils.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+func combineLowHigh(tokenIdLowHex, tokenIdHighHex string) (*big.Int, error) {
+	// Remove leading zeros from the hex strings
+	tokenIdLowHex = strings.TrimLeft(tokenIdLowHex, "0")
```
Comment (b-j-roberts) [inline backend/routes/indexer/utils.go:11]: Please change all variable naming away from tokenId now that the function is generic

File: backend/routes/indexer/utils.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	tokenIdHighHex = strings.TrimLeft(tokenIdHighHex, "0")
```
Comment (b-j-roberts) [inline backend/routes/indexer/utils.go:12]: You wont want to trim all the left 16 0s that are unused.

For example :
High: `00000000000000000000000000000015`
Low:  `00000000000000000000000000001234`
would give `151234`, which is incorrect. It should give `150000000000001234`

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: backend/routes/indexer/utils.go (hunk 4 above)
Comment (manlikeHB) [reply backend/routes/indexer/utils.go:12]: oh! my bad, I'll fix it now.

**Comment by b-j-roberts** [comment]
Thank you

**Pull request merged**